iac-extensions/docker/src/test/java/org/sonar/iac/docker/checks/AddAndCopyPermissionCheckTest.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.checks;
+
+import org.junit.jupiter.api.Test;
+
+class AddAndCopyPermissionCheckTest {
+
+  @Test
+  void test_add() {
+    DockerVerifier.verify("AddAndCopyPermissionCheckTest/Dockerfile_add", new AddAndCopyPermissionCheck());
+  }
+
+  @Test
+  void test_copy() {
+    DockerVerifier.verify("AddAndCopyPermissionCheckTest/Dockerfile_copy", new AddAndCopyPermissionCheck());
+  }
```

Comment:
I'm not sure if a dedicated test is nesseary. This only increases the maintenance effort. I would keep only one test and show, with 1-2 lines if `COPY` that there is no difference between `ADD` and `COPY`.